### PR TITLE
feat(sort): add `StreamSort` plan node

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- ~~[ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- ~~[ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/stream_sort.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    fn infer_state_table(&self) -> TableCatalog {
+        // The sort state table has the same schema as the input.
+
+        let in_fields = self.input.schema().fields();
+        let mut tbl_builder =
+            TableCatalogBuilder::new(self.ctx().with_options().internal_table_subset());
+        for field in in_fields {
+            tbl_builder.add_column(field);
+        }
+        tbl_builder.add_order_column(self.sort_column_index, OrderType::ascending());
```
Comment (soundOfDestiny) [inline src/frontend/src/optimizer/plan_node/stream_sort.rs:77]: Do we need to add the original pk here?

**Review by soundOfDestiny — APPROVED** [review]
LGTM

File: src/frontend/src/optimizer/plan_node/stream_sort.rs (hunk 1 above)
Comment (richardchien) [reply src/frontend/src/optimizer/plan_node/stream_sort.rs:77]: Oh yes, I missed🥵

File: src/frontend/src/optimizer/plan_node/stream_sort.rs [hunk 2]
```diff
@@ -75,6 +75,9 @@ impl StreamSort {
             tbl_builder.add_column(field);
         }
         tbl_builder.add_order_column(self.sort_column_index, OrderType::ascending());
+        for &pk_idx in self.input.logical_pk() {
+            tbl_builder.add_order_column(pk_idx, OrderType::ascending());
```
Comment (soundOfDestiny) [inline src/frontend/src/optimizer/plan_node/stream_sort.rs:79]: better dedup here

**Review by soundOfDestiny — COMMENTED** [review]
or we might dedup in `add_order_column`

File: src/frontend/src/optimizer/plan_node/stream_sort.rs (hunk 2 above)
Comment (richardchien) [reply src/frontend/src/optimizer/plan_node/stream_sort.rs:79]: OK

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Pull request merged**